lib/medloaders/brats2020.py

```diff
@@ -0,0 +1,120 @@
+import glob
+import os
+
+import numpy as np
+import torch
+from torch.utils.data import Dataset
+
+import lib.augment3D as augment3D
+import lib.utils as utils
+from lib.medloaders import medical_image_process as img_loader
+from lib.medloaders.medical_loader_utils import create_sub_volumes
+
+
+class MICCAIBraTS2020(Dataset):
+    """
+    Code for reading the infant brain MICCAIBraTS2018 challenge
+    """
+
+    def __init__(self, args, mode, dataset_path='./datasets', classes=5, crop_dim=(200, 200, 150), split_idx=260,
+                 samples=10,
+                 load=False):
+        """
+        :param mode: 'train','val','test'
+        :param dataset_path: root dataset folder
+        :param crop_dim: subvolume tuple
+        :param split_idx: 1 to 10 values
+        :param samples: number of sub-volumes that you want to create
+        """
+        self.mode = mode
+        self.root = str(dataset_path)
+        self.training_path = self.root + '/brats2020/MICCAI_BraTS_2020_Data_Training/'
+        self.testing_path = self.root + '/brats2020/MICCAI_BraTS_2020_Data_Validation/'
+        self.full_vol_dim = (240, 240, 155)  # slice, width, height
+        self.crop_size = crop_dim
+        self.threshold = args.threshold
+        self.normalization = args.normalization
+        self.augmentation = args.augmentation
+        self.list = []
+        self.samples = samples
+        self.full_volume = None
+        self.classes = classes
+        if self.augmentation:
+            self.transform = augment3D.RandomChoice(
+                transforms=[augment3D.GaussianNoise(mean=0, std=0.01), augment3D.RandomFlip(),
+                            augment3D.ElasticTransform()], p=0.5)
+        self.save_name = self.root + '/brats2020/brats2020-list-' + mode + '-samples-' + str(samples) + '_size_' + str(
+            crop_dim[0]) + '.txt'
+
+        if load:
+            ## load pre-generated data
+            self.list = utils.load_list(self.save_name)
+            list_IDsT1 = sorted(glob.glob(os.path.join(self.training_path, '*GG/*/*t1.nii.gz')))
+            self.affine = img_loader.load_affine_matrix(list_IDsT1[0])
+            return
+
+        subvol = '_vol_' + str(crop_dim[0]) + 'x' + str(crop_dim[1]) + 'x' + str(crop_dim[2])
```

Comment:
f-strings instead of `+`: `f '_vol_ {str(crop_dim[0])}x{ str(crop_dim[1])}x{str(crop_dim[2])}'`